pkg/envoy/server.go

```diff
@@ -855,32 +902,43 @@ func getDirectionNetworkPolicy(l4Policy policy.L4PolicyMap, policyEnforced bool)
 		pnp := &cilium.PortNetworkPolicy{
 			Port:     uint32(l4.Port),
 			Protocol: protocol,
-			Rules:    make([]*cilium.PortNetworkPolicyRule, 0, len(l4.L7RulesPerEp)),
+			Rules:    make([]*cilium.PortNetworkPolicyRule, 0, len(l4.L7RulesPerSelector)),
 		}
-
 		allowAll := false
 		// Assume none of the rules have side-effects so that rule evaluation can
 		// be stopped as soon as the first allowing rule is found. 'canShortCircuit'
 		// is set to 'false' below if any rules with side effects are encountered,
 		// causing all the applicable rules to be evaluated instead.
 		canShortCircuit := true
-		for sel, l7 := range l4.L7RulesPerEp {
-			rule, cs := getPortNetworkPolicyRule(sel, l4.L7Parser, l7)
-			if !cs {
-				canShortCircuit = false
-			}
+
+		if l4.Port == 0 {
+			// Wildcard L4 rule, must generate L7 allow-all in case there are other
```

Comment:
nit: 
```suggestion
			// L3-only rule, must generate L7 allow-all in case there are other
```

I feel like we've always talked about l3-only, l4-only, l3-dependent-l7 (L3+L4+L7), l4-dependent-l7 (L4+L7) as the standard way to think about these rules.

Reply:
Will change.